File: KotlinAndroid/AsHttp/src/main/java/org/lxz/utils/http/HttpException.java
package org.lxz.utils.http;

import java.io.IOException;

/**
 * Created by Lin on 2017/4/21.
 */

public class HttpException extends IOException {
    private int code;
    private String msg;

    public HttpException(int code, String message) {
        this.code = code;
        this.msg = message;
    }

    public HttpException(Throwable e) {
        code=-1;
        msg=e.getMessage();
    }

    public int getCode() {
        return code;
    }

    public HttpException setCode(int code) {
        this.code = code;
        return this;
    }


    @Override
    public String getMessage() {
        return msg;
    }


    @Override
    public String toString() {
        return  msg ;
    }
}

File: KotlinAndroid/kotlinlibrary/src/main/java/com/lxz/kotlin/tools/http/HttpException.kt
package com.lxz.kotlin.tools.http

import java.io.IOException

/**
 * Created by linxingzhu on 2018/1/9.
 */
class HttpException : IOException {
    private var code: Int = 0
    private var msg: String

    constructor(code: Int, message: String) {
        this.code = code
        this.msg = message
    }

    constructor(e: Throwable) {
        code = -1
        msg = e.message!!
    }

    fun getCode(): Int {
        return code
    }

    fun setCode(code: Int): HttpException {
        this.code = code
        return this
    }


    fun getMsg(): String{
        return msg
    }


    override fun toString(): String{
        return msg
    }
}